features/catalog/step_definitions/descriptor-deletion.ts

```diff
@@ -22,32 +22,35 @@ When("l'utente cancella il descrittore di quell'e-service", async function () {
   );
 });
 
-Then("il descrittore è stato cancellato, ma l'eservice no", async function () {
-  assertContextSchema(this, {
-    token: z.string(),
-    eserviceId: z.string(),
-    descriptorId: z.string(),
-  });
+Then(
+  "il descrittore è stato cancellato, e anche l'eservice",
+  async function () {
+    assertContextSchema(this, {
+      token: z.string(),
+      eserviceId: z.string(),
+      descriptorId: z.string(),
+    });
 
-  await makePolling(
-    () =>
-      apiClient.producers.getProducerEServiceDescriptor(
-        this.eserviceId,
-        this.descriptorId,
-        getAuthorizationHeader(this.token)
-      ),
-    (res) => res.status === 404
-  );
+    await makePolling(
+      () =>
+        apiClient.producers.getProducerEServiceDescriptor(
+          this.eserviceId,
+          this.descriptorId,
+          getAuthorizationHeader(this.token)
+        ),
+      (res) => res.status === 404
+    );
 
-  await makePolling(
-    () =>
-      apiClient.producers.getProducerEServiceDetails(
-        this.eserviceId,
-        getAuthorizationHeader(this.token)
-      ),
-    (res) => res.status !== 404
-  );
-});
+    await makePolling(
+      () =>
+        apiClient.producers.getProducerEServiceDetails(
+          this.eserviceId,
+          getAuthorizationHeader(this.token)
+        ),
+      (res) => res.status === 404
```

Comment:
Since here we delete the whole eservice, we could just poll on the eservice till it's 404

Reply:
I prefer to keep those as clear as possible